src/VirtoCommerce.Platform.Web/Controllers/ExternalSignInController.cs

```diff
@@ -82,13 +82,15 @@ public async Task<ActionResult> SignInCallback(string returnUrl)
                 //Need handle the two cases
                 //first - when the VC platform user account already exists, it is just missing an external login info and
                 //second - when user does not have an account, then create a new account for them
-                platformUser = await _userManager.FindByNameAsync(userName);
+                platformUser = await _userManager.FindByNameAsync(userName) ??
+                               await FindUserByEmail(GetUserEmail(externalLoginInfo));
+
                 if (platformUser == null)
                 {
                     platformUser = new ApplicationUser
                     {
                         UserName = userName,
-                        Email = externalLoginInfo.Principal.FindFirstValue(ClaimTypes.Email) ?? (userName.IsValidEmail() ? userName : null),
+                        Email = GetUserEmail(externalLoginInfo) ?? (userName.IsValidEmail() ? userName : null),
```

Comment:
What is the point to move the previous statement to the private method?

Reply:
It is made to avoid code duplication.